subprojects/language-native/src/integTest/groovy/org/gradle/language/swift/SwiftExecutableIntegrationTest.groovy

```diff
@@ -63,6 +77,61 @@ class SwiftExecutableIntegrationTest extends AbstractInstalledToolChainIntegrati
         installation("build/install/App").exec().out == app.expectedOutput
     }
 
+    def "stalled object files are removed"() {
+        settingsFile << "rootProject.name = 'app'"
+        def app = new SwiftApp()
+
+        given:
+        app.writeToProject(testDirectory)
+
+        and:
+        buildFile << """
+            apply plugin: 'swift-executable'
+         """
+
+        and:
+        succeeds "assemble"
+        file(app.multiply.sourceFile.withPath("src/main")).delete()
+        file(app.greeter.sourceFile.withPath("src/main")).renameTo(file("src/main/swift/renamed-greeter.swift"))
+
+        expect:
+        succeeds "assemble"
+        result.assertTasksExecuted(":compileSwift", ":linkMain", ":installMain", ":assemble")
+        result.assertTasksNotSkipped(":compileSwift", ":linkMain", ":installMain", ":assemble")
+
+        file("build/main/objs").eachFileRecurse(FileType.FILES) {
+            assert it.name != app.multiply.sourceFile.name.replace('.swift', '.o')
```

Comment:
I wonder if we should also be looking for stale `.siwftdoc` and `.swiftmodule` files as well?